charts/agent-control/templates/test.yaml

```diff
@@ -0,0 +1,81 @@
+{{- $testPodName := include "newrelic.common.naming.truncateToDNSWithSuffix" ( dict "name" (include "newrelic.common.naming.fullname" .) "suffix" "helm-test" ) -}}
+apiVersion: v1
+kind: Pod
+metadata:
+  name: {{ $testPodName }}
+  namespace: {{ .Release.Namespace }}
+  labels:
+    {{- include "newrelic.common.labels" . | nindent 4 }}
+  annotations:
+    helm.sh/hook: test
+    # removing the pod after the hook prevent to fetch the logs from helm cli
+    helm.sh/hook-delete-policy: before-hook-creation
+spec:
+  restartPolicy: Never
+  serviceAccountName: {{ $testPodName }}
+  containers:
+  - name: log-scraper
+    image: {{ include "newrelic.common.images.image" ( dict "imageRoot" .Values.toolkitImage "context" .) }}
+    imagePullPolicy: {{ .Values.toolkitImage.pullPolicy }}
+    resources:
+      limits:
+        cpu: 100m
+        memory: 128Mi
+      requests:
+        cpu: 50m
+        memory: 64Mi
+    command: ["/bin/sh", "-c"]
+    args:
+      - |
+        # This is not a test but a workaround to print the installations Jobs logs in case of failure to provide a better 
+        # feedback for the user.
+        # Jobs are executed with Helm hooks and configured so the Pods don't get removed in case of failure.
+        # Helm V4 is already providing an alternative https://github.com/helm/community/blob/main/hips/hip-0019.md#specification
+
+        kubectl logs -l batch.kubernetes.io/job-name --prefix --tail=-1 -n {{ .Release.Namespace }}
+        
+        # This is not a test
+        exit 0
+
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
```

Comment:
I tried to add hooks for removing these rbac resources , but they were consider as different tests , and the output of the helm command was repeating the pod output multiple times. I even try to filter them with --filter but didn't work. 
I didn't dig into helm's code as i thought that these are pretty low important permissions , so keep them installed is not an issue.

Reply:
oh, that is super weird, also the service account it "kept", no?

Reply:
even with the same name "helm.sh/hook: test"?